### PR TITLE
CCP-2151 swatches for PLP

### PR DESCRIPTION
**Pull request by Carsten89**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,2 +1,32 @@
-# fashion-swatches
-Adds color swatches to PLP and PDP. Shows configured attributes fashion optimized on PDP 
+# Shopgate Connect - Fashion swatches
+
+This extension adds color swatches to PLP and PDP. Configured attributes/characteristics are shown in a fashio optimized styling on PDP.
+
+## Configuration
+### swatchStylePlp (json)
```
Comment (alexbridge) [inline README.md]: Can we re-organize a config to have smthng like
```
{
  "colorSwatches": {
    "default": {
      "swatchStylePlp": {
        "borderRadius": "100%",
        "border": "2px solid black"
      }
    },
    "plp": {
      "swatchStylePlp": {
        "borderRadius": "100%",
        "border": "2px solid black"
      }
    },
    "pdp": {
      "swatchStylePlp": {
        "borderRadius": "100%",
        "border": "2px solid black"
      }
    }
  },
  "propertyWithColors": "Foo-bar"
}
```

File: extension-config.json [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+  "components": [
+    {
+      "id": "PlpSwatches",
+      "path": "frontend/components/PlpSwatches/index.jsx",
+      "target": "product-item.price.after",
+      "type": "portals"
+    }
+  ],
+  "configuration": {
+    "swatchStylePlp": {
```
Comment (alexbridge) [inline extension-config.json]: Check a comment about re-organizing a config

File: frontend/selectors/index.js [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+export const getPlpSwatches = createSelector(
+  getProductDataById,
+  (productData) => {
+    const { additionalProperties = {} } = productData || {};
+
+    if (!additionalProperties[propertyWithColors]) {
+      return null;
+    }
+
+    const swatches = JSON.parse(additionalProperties[propertyWithColors]);
```
Comment (alexbridge) [inline frontend/selectors/index.js]: Add a try catch block. It can through Parse exception

File: frontend/selectors/index.js [hunk 4]
```diff
@@ -0,0 +1,24 @@
...
+
+    if (!Array.isArray(swatches)) {
+      logger.warn(`property "${propertyWithColors}" has wrong format. Expected json string, got`, additionalProperties[propertyWithColors]);
+      return null;
+    }
+
+    return JSON.parse(additionalProperties[propertyWithColors]);
```
Comment (alexbridge) [inline frontend/selectors/index.js]: You alreeady parsed it on line 15

File: frontend/components/PlpSwatches/style.js [hunk 5]
```diff
@@ -0,0 +1,35 @@
...
+const list = css({
+  display: 'flex',
+  flexDirection: 'row',
+  flexWrap: 'wrap',
+  marginTop: 5,
+});
+
+const listItem = css({
+  marginRight: 5,
+  marginBottom: 10,
```
Comment (alexbridge) [inline frontend/components/PlpSwatches/style.js]: I guess 5 as bottom margin would look better

**Review by alexbridge — CHANGES_REQUESTED** [review]

File: extension-config.json [hunk 6]
```diff
@@ -0,0 +1,38 @@
...
+      "type": "admin",
+      "destination": "frontend",
+      "default": {
+        "borderRadius": "100%",
+        "border": "2px solid black"
```
Comment (alexbridge) [inline extension-config.json]: Sketch had no border: https://jira.shopgate.guru/secure/attachment/58957/Frame%206.jpg

**Review by alexbridge — CHANGES_REQUESTED** [review]

File: extension-config.json (hunk 6 above)
Comment (Carsten89) [reply extension-config.json]: testing leftover. will remove default

**Review by alexbridge — APPROVED** [review]

**Pull request merged**